engine/schema/src/main/resources/META-INF/db/schema-41310to41400.sql

```diff
@@ -379,3 +379,81 @@ CREATE TABLE IF NOT EXISTS `cloud`.`kubernetes_cluster_details` (
     PRIMARY KEY(`id`),
     CONSTRAINT `fk_kubernetes_cluster_details__cluster_id` FOREIGN KEY `fk_kubernetes_cluster_details__cluster_id`(`cluster_id`) REFERENCES `kubernetes_cluster`(`id`) ON DELETE CASCADE
 ) ENGINE=InnoDB DEFAULT CHARSET=utf8;
+
+ALTER TABLE `cloud`.`roles` ADD COLUMN `is_default` tinyint(1) NOT NULL DEFAULT '0' COMMENT 'is this a default role';
+UPDATE `cloud`.`roles` SET `is_default` = 1 WHERE id IN (1, 2, 3, 4);
```

Comment:
Please check in code if there exists any restriction on ID for default/built-in roles.

Reply:
> Please check in code if there exists any restriction on ID for default/built-in roles.

The role ids (1,2,3,4) are hard-coded when upgraded to dynamic roles access checker. The same roles are now marked as default roles in cloudstack.